src/core/Sampler/Sampler.cpp

```diff
@@ -257,6 +280,30 @@ bool Sampler::renderNote( Note* pNote, unsigned nBufferSize, Song* pSong )
 		return 1;
 	}
 
+	// new instrument and note pan interaction--------------------------
+	// note pan moves the pan in a smaller pan range centered at instrument pan
+
+	// get the note and instrument pan parameters in [-1,1]
+	// TODO a new song member, set in preferences, must point the desired pan parameter Sampler member function
+	float (*getPan)( float, float );
+	// use "ratio" pan parameter (due to the current pan law)
+    getPan = &this->getRatioPan;
```

Comment:
Why don't you use `getRatioPan` straight away?

Reply:
Not all the pan laws use that definition of the parameter.
In the standard adopted linear panning (a particular case of constant sum pan law i.e. pan_L + pan_R = const) for example the pan parameter has a different dependence on pan_L and pan_R.
So ideally another `getLinearPan( float, float)` function could be pointed

Reply:
You made me think that the implementation could **always** use the method `ratioStraightPolPanLaw(pan)` to save `pan_L` and `pan_R` and **always** `getRatioPan(pan_L, pan_R)` to reconvert them to the initial single parameter. So all this is just a fixed bijection, that we use only to keep unchanged the way h2 stores the pan parameter information. Even if this will be less intuitive for developers.
Then, when rendering the note, the single pan parameter can be interpreted differently depending on the pan law.
So there is no need of `getLinearPan( float, float)` or `getLinearPan( float, float)`.